eventuate-tram-sagas-reactive-orchestration/src/main/java/io/eventuate/tram/sagas/reactive/orchestration/ReactiveSagaInstanceRepositoryJdbc.java

```diff
@@ -96,10 +95,7 @@ public ReactiveSagaInstanceRepositoryJdbc(EventuateReactiveJdbcStatementExecutor
                         new SerializedSagaData(row.get("saga_data_type").toString(), row.get("saga_data_json").toString()), new HashSet<>(dar)),
               sagaType, sagaId).collectList());
 
-    return result.flatMap(sagaInstances -> {
-      if (sagaInstances.isEmpty()) return Mono.error(new RuntimeException(String.format("Cannot find saga instance %s %s", sagaType, sagaId)));
-      else return Mono.just(sagaInstances.get(0));
-    });
+    return result.flatMapMany(Flux::fromIterable).single();
```

Comment:
Why not simply `eventuateJdbcStatementExecutor.queryForList(...).single()`?

Reply:
Fixed and tested: https://github.com/eventuate-tram/eventuate-tram-sagas/pull/64